clang-tidy-checks/potential-unbounded-loop-check.diff

```diff
@@ -0,0 +1,3051 @@
++        _gencode_helper("<=", "++", t2, t1),
++        _gencode_helper(">", "--", t2, t1),
++        _gencode_helper(">=", "--", t2, t1),
++        _gencode_helper(">", "++", t2, t1, swap_comp_order=True),
++        _gencode_helper(">=", "++", t2, t1, swap_comp_order=True),
++        _gencode_helper("<", "--", t2, t1, swap_comp_order=True),
++        _gencode_helper("<=", "--", t2, t1, swap_comp_order=True),
++        _gencode_helper("<", "++", t1, t2, trigger=True),
++        _gencode_helper("<=", "++", t1, t2, trigger=True),
++        _gencode_helper(">", "--", t1, t2, trigger=True),
++        _gencode_helper(">=", "--", t1, t2, trigger=True),
++        # swap comparision order for t1 and t2
++        # ensures that a potential unbounded loop exists
++        _gencode_helper(">", "++", t1, t2, trigger=True, swap_comp_order=True),
++        _gencode_helper(">=", "++", t1, t2, trigger=True, swap_comp_order=True),
++        _gencode_helper("<", "--", t1, t2, trigger=True, swap_comp_order=True),
++        _gencode_helper("<=", "--", t1, t2, trigger=True, swap_comp_order=True),
++    ]
++
++    return "".join(tests)
++
++
++def gencode_eq(t1):
++    """Generates tests using single loop var type
++
++    Arguments:
++        t1  Integer type
++    """
++    tests = [
++        _gencode_helper("<", "++", t1, t1),
++        _gencode_helper("<=", "++", t1, t1),
++        _gencode_helper(">", "--", t1, t1),
++        _gencode_helper(">=", "--", t1, t1),
++    ]
++
++    return "".join(tests)
++
++
++def gen(test):
++    _, types = test
++    n = len(types)
++
++    code = []
++
++    for i in range(n):
++        code.append(gencode_eq(types[i]))
++        for j in range(i + 1, n):
++            code.append(gencode_lt(types[i], types[j]))
++
++    return "".join(code)
++
++
++def main():
++    test_suite = {
++        "signed_int_types": ["int8_t", "int16_t", "int32_t", "int64_t"],
++        "unsigned_int_types": ["uint8_t", "uint16_t", "uint32_t", "uint64_t"],
++    }
++
++    preamble = textwrap.dedent(
++        f"""\
++        // @generated clang-tools-extra/clang-tidy/misc/gen_potential_unbounded_loop_test.py
++        // RUN: %check_clang_tidy %s {CHECK_NAME} %t
++    """
++    )
++
++    code = ["#include<stdint.h>\n"]
++
++    for test in test_suite.items():
++        code.append(gen(test))
++
++    print(f'{preamble} {"".join(code)}')
++
++
++if __name__ == "__main__":
++    main()
+diff --git a/clang-tools-extra/clang-tidy/rename_check.py b/clang-tools-extra/clang-tidy/rename_check.py
+index 4d5311c9a293..2410041fd5d2 100755
+--- a/clang-tools-extra/clang-tidy/rename_check.py
++++ b/clang-tools-extra/clang-tidy/rename_check.py
+@@ -120,16 +120,15 @@ def adapt_cmake(module_path, check_name_camel):
+       if (not file_added) and (cpp_line or cpp_found):
+         cpp_found = True
+         if (line.strip() > cpp_file) or (not cpp_line):
+-          f.write('  ' + cpp_file + '\n')
++          f.write(('  ' + cpp_file + '\n').encode())
```

Comment:
This does not seem related to the patch, does it?

Reply:
I used this script to rename the check, but the script was buggy. Upstream LLVM had a fixed version, which I imported. I agree, it might be better to have a separate patch for this.